syntax = "proto3";

package transport_catalogue_proto;

import "map_renderer.proto";
import "transport_router.proto";

message Coords {
    double lat = 1;
    double lng = 2;
}

message Stop {
    string stop_name = 1;
    Coords coords = 2;
    uint32 id = 3;
}

message Bus {
    string bus_num = 1;
    bool is_circle = 2;
    repeated uint32 stop_indexes = 3;
}

message Distance {
    uint32 from = 1;
    uint32 to = 2;
    int32 distance = 3;
}

message TransportCatalogue {
    repeated Stop stops = 1;
    repeated Bus buses = 2;
    repeated Distance distances = 3;
    RenderSettings render_settings = 4;
    BusTimesSettings time_settings = 5;
}